.github/workflows/main.yaml

```diff
@@ -0,0 +1,94 @@
+name: basic tests
+
+on:
+  push:
+    branches:
+      - 'user/**'
+      - 'feature/**'
+      - 'improvement/**'
+      - 'bugfix/**'
+      - 'w/**'
+      - 'q/**'
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildk
+        uses: docker/setup-buildx-action@v1
+        with:
+          buildkitd-flags: --debug
+
+      - name: Login to Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          context: .github/docker
+          file: .github/docker/Dockerfile
+          build-args: BUILDBOT_VERSION=2.0.1
+          tags: "ghcr.io/scality/bert-e:${{ github.sha }}"
+
+  tests:
+    runs-on: ubuntu-latest
+    container:
+      image: "ghcr.io/scality/bert-e:${{ github.sha }}"
+      options: --user=0
+    needs:
+      - publish
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - run: tox -e utests
+      - run: tox -e tests-api-mock
+      - run: tox -e coverage-report
+      - run: tox -e tests-server
+
+  tests-noqueue:
+    runs-on: ubuntu-latest
+    container:
+      image: "ghcr.io/scality/bert-e:${{ github.sha }}"
```

Comment:
So ghcr.io is a new kind of docker repo ? How is that handled ? how do we manage image on it ?

Reply:
Ah ha, good question, that's github docker registry that comes with the "package".

But this make me wonder, do we want another registry? Should we rather deactivate it? My answer for now is I don't want another registry. so I wonder if we should rather change that url to registry.scality.com and that would actually be a good test to do, to make sure actions can work with that registry.

For bert-e the image would be public on registry.scality.com, so no issue there. But for other repos I believe we may find an issue where actions may not be capable of directly pulling a private image. :thinking: I'm gonna write this one down to test.

Thanks @cmeron-scality!

- [ ] Change registry url to registry.scality.com (@erwan-b let's work together on this one tomorrow)

Reply:
I challenge this request and I am not fully conviced we should mix github action CI and scality tools if we can avoid it. I think we should use github tools as much as possible in the way it was meant to be used. This will probably lead to a more natural and maintainable integration.